internal/core/cppapi_data.cpp

```diff
@@ -417,4 +417,54 @@ void attributes(
     calc_attributes(horizon, surface, src_window, dst_window, attrs, from, to);
 }
 
+void align_surfaces(
```

Comment:
What about taking in RegularSurface as output variable instead?
```
void align_surfaces(
     RegularSurface const &primary,
     RegularSurface const &secondary,
     RegularSurface & aligned, 
     bool *primary_is_top
)
```

Then align_surfaces would update the values and the plane definition - i.e. producing the full aligned surfaces, ready to use. 

With a signature like that we can even update secondary in place if we don't need it again:
```
aling_surfaces(primary, secondary, secondary, *istop);
```